genvalidity-hspec/src/Test/Validity/Utils.hs

```diff
@@ -68,7 +69,11 @@ instance Functor Anon where
 -- but it has been copied here for convenience.
 -- https://github.com/hspec/hspec/commit/020c7ecc4a73c24af38e9fab049f60bb9aec6981#diff-29cb22f0ef6e98086a71fc045847bd21L22
 mapSpecTree' :: (SpecTree a -> SpecTree b) -> SpecM a r -> SpecM b r
+#if MIN_VERSION_hspec(2,10,0)
+mapSpecTree' f (SpecM specs) = SpecM (mapWriterT (fmap (second (fmap (map f)))) specs)
+#else
 mapSpecTree' f (SpecM specs) = SpecM (mapWriterT (fmap (second (map f))) specs)
+#endif
 
 -- | Asserts that a given 'Spec' tree fails _somewhere_.
 --
```

Comment:
I think you don’t really need `mapSpecTree'` to implement `failsBecause`.

Something like this should work:

```haskell
failsBecause :: HasCallStack => String -> SpecWith () -> SpecWith ()
failsBecause s sp = it s $ do
    let conf = defaultConfig {configFormatter = Just silent}
    r <- hspecWithResult conf sp
    let succesful = summaryExamples r > 0 && summaryFailures r > 0
    throwIO $ produceResult succesful
```

Reply:
@NorfairKing the only difference here is that the spec item will have a location for the call site of `failsBecause`. That’s usually what you want. If it is not suitable for some reason then remove the location with `mapSpecItem_`.

Doing it that way is future proof and does not require `CPP`.

Reply:
`failsBecause` can be removed IMO, if that would help.